src/rasterstats/main.py

```diff
@@ -201,12 +296,34 @@ def gen_zonal_stats(
                 if 'max' in stats:
                     feature_stats['max'] = float(masked.max())
                 if 'mean' in stats:
-                    feature_stats['mean'] = float(masked.mean())
+                    if percent_cover_weighting and latitude_correction:
+                        feature_stats['mean'] = float(
+                            np.sum((masked.T * latitude_scale).T * cover_weights) /
+                            np.sum((~masked.mask.T * latitude_scale).T * cover_weights))
+                    elif percent_cover_weighting:
+                        feature_stats['mean'] = float(
+                            np.sum(masked * cover_weights) /
+                            np.sum(~masked.mask * cover_weights))
+                    elif latitude_correction:
+                        feature_stats['mean'] = float(
+                            np.sum((masked.T * latitude_scale).T) /
+                            np.sum(latitude_scale *
+                                   (masked.shape[1] - np.sum(masked.mask, axis=1))))
+                    else:
+                        feature_stats['mean'] = float(masked.mean())
+
```

Comment:
The latitude correction and the percent cover add branching complexity to the code and it detracts from readability a bit. Let's implement this with less `if` statements to test.

Reply:
sounds good